src/ray/stats/metric_exporter.cc

```diff
@@ -106,8 +101,9 @@ void MetricExporter::ExportViewData(
       break;
     }
   }
-  RAY_LOG(DEBUG) << "Point size : " << points.size();
+
   metric_exporter_client_->ReportMetrics(points);
+  metric_exporter_client_->ReportMetrics(data);
```

Comment:
ReportMetrics interface should keep two input parameters, one for MetricPoints and other for original viewdata. I remember that we talked in slack.

Reply:
Hi @ashione, here are more context. If we define `ReportMetrics(MetricPoints, OpenCensusData)`. The issue here are two folds:
- MetricPoints are derived from OpenCensusData. This means if we only need OpenCensus data, the MetricsPoints computation is redundant. (For the MetricExporterClient implementations, the `ReportMetrics` are implemented in Either-Or.)
- `MetricPoints` has different length than `OpenCensusData` because the distribution values from `OpenCensusData` are expanded into max, min, and average. This makes it difficult for batching.

Additionally, the MetricExporterClient makes it possible to just override one method: `ReportMetrics(MetricPoints)` and leave the OpenCensus on unimplemented. This marks it clear the `MetricPoints` are the only implementation required for all sublcasses and use the decorator chain to pass on the data. `ReportMetrics(OpenCensusData)` is entirely optionally and fall back to empty implementation block (Noop with a warning). `ReportMetrics(OpenCensusData)` will not go through the decorator chain.

Reply:
> Hi @ashione, here are more context. If we define `ReportMetrics(MetricPoints, OpenCensusData)`. The issue here are two folds:
> 
> * MetricPoints are derived from OpenCensusData. This means if we only need OpenCensus data, the MetricsPoints computation is redundant. (For the MetricExporterClient implementations, the `ReportMetrics` are implemented in Either-Or.)
> * `MetricPoints` has different length than `OpenCensusData` because the distribution values from `OpenCensusData` are expanded into max, min, and average. This makes it difficult for batching.
> 
> Additionally, the MetricExporterClient makes it possible to just override one method: `ReportMetrics(MetricPoints)` and leave the OpenCensus on unimplemented. This marks it clear the `MetricPoints` are the only implementation required for all sublcasses and use the decorator chain to pass on the data. `ReportMetrics(OpenCensusData)` is entirely optionally and fall back to empty implementation block (Noop with a warning). `ReportMetrics(OpenCensusData)` will not go through the decorator chain.

Yes. I know the ReportMetrics(metricPoints) will split view data into different pieces so it's hard to unify them to one interface.

Reply:
@rkooo567 Have you any suggestions about that?

Reply:
I think the best solution here is just not process data inside `ExportViewData` and do everything in the reporter client. We can split reporter client into two to avoid replicating data processing logic (ex, MetricsOCClient, MetricsPointClient). I think suggested solution is also a clean solution.